GenerateOppMessage.cmake

```diff
@@ -1,31 +1,97 @@
 include(CMakeParseArguments)
 
+# Options:
+# - MSG4: Call message compiler with --msg4
+# 
+# One Value Arguments:
+# - TARGET: Optional name of the target to call target_sources and target_include_directories to add compiled sources
+# - OUTPUT_ROOT: Optional root-directory where the files will be generated. Default: ${PROJECT_BINARY_DIR}/opp_messages
+# - DIRECTORY: Optional relative path for the output-directory (Appended to OUTPUT_ROOT). 
+#   Defaults to the relative path of the message file in respect to ${PROJECT_SOURCE_DIR}/src
+# - GEN_SOURCES: Optional name of the variable to populate with sources to be compiled
+# - GEN_INCLUDE_DIR: Optional name of the variable to populate with the include directory
+#
+# Mutli Value Arguments:
+# - ADDITIONAL_NED_PATHS: Optional paths to be added as search/include directories when calling the message compiler (-I Arguments)
+
 # generate sources for messages via opp_msgc
 function(generate_opp_message msg_input)
-    cmake_parse_arguments(args "" "TARGET;DIRECTORY" "" ${ARGN})
+    set(options MSG4)
+    set(oneValueArgs TARGET DIRECTORY OUTPUT_ROOT GEN_SOURCES GEN_INCLUDE_DIR)
+    set(multiValueArgs ADDITIONAL_NED_PATHS)
+
+    cmake_parse_arguments(args "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
```

Comment:
```suggestion
    set(options_args MSG4)
    set(single_args TARGET DIRECTORY OUTPUT_ROOT GEN_SOURCES GEN_INCLUDE_DIR)
    set(multi_args ADDITIONAL_NED_PATHS)

    cmake_parse_arguments(args "${options_args}" "${single_args}" "${multi_args}" ${ARGN})
```

**suggestion (non-blocking)**: use the same names for the argument variables as in `add_opp_target`. 

I wouldn't mind calling them something else entirely, although I'm not sure what I'd suggest. The bikeshedding in itself isn't too important. Consistency would be nice, however.